pkg/ensurance/collector/noderesource/noderesource.go

```diff
@@ -0,0 +1,63 @@
+package noderesource
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	v1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/gocrane/crane/pkg/common"
+	"github.com/gocrane/crane/pkg/ensurance/collector/types"
+	"github.com/gocrane/crane/pkg/utils"
+)
+
+type NodeResource struct {
+	nodeName   string
+	nodeLister v1.NodeLister
+	podLister  v1.PodLister
+}
+
+func NewNodeResourceCollector(nodeName string, nodeLister v1.NodeLister, podLister v1.PodLister) *NodeResource {
+	klog.V(4).Infof("NodeResourceCollector create")
+	return &NodeResource{
+		nodeName:   nodeName,
+		nodeLister: nodeLister,
+		podLister:  podLister,
+	}
+}
+
+func (n *NodeResource) GetType() types.CollectType {
+	return types.NodeResourceCollectorType
+}
+
+func (n *NodeResource) Collect() (map[string][]common.TimeSeries, error) {
+	klog.V(4).Infof("NodeResourceCollector Collect")
+	node, err := n.nodeLister.Get(n.nodeName)
+	if err != nil {
+		return nil, err
+	}
+	pods, err := n.podLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	allExtCpu := node.Status.Allocatable.Name(corev1.ResourceName(fmt.Sprintf(utils.ExtResourcePrefixFormat, corev1.ResourceCPU.String())), resource.DecimalSI).MilliValue()
```

Comment:
if the Allocatable not have extend resource, when does the agent start, it will crash or not?

Reply:
if the Allocatable not have extend resource, node.Status.Allocatable.Name returns an empty Quantity, it will not crash.